src/Components/SmartComponents/SystemsExposedTable/SystemsExposedTable.js

```diff
@@ -1,410 +1,303 @@
-        handleCveDetailsError: propTypes.func,
-        createNotification: propTypes.func,
-        selectRows: propTypes.func,
-        selectMultiple: propTypes.func,
-        intl: propTypes.any,
-        onCollapse: propTypes.func
+const SystemsExposedTable = (props) => {
+    const [InventoryTable, setInventoryTable] = useState(()=> () => <div>Loading...</div>);
+    const [filter, setFilter] = useState(qs.parse(props.location.search).filter);
+    const [StatusModal, setStatusModal] = useState(()=>null);
+    const [selectedHosts, setSelectedHosts] = useState(undefined);
+    const [statusId, setStatusId] = useState(undefined);
+    const [securityRule, setSecurityRule] = useState(undefined);
+    const [page] = useState(1);
+    const [page_size] = useState(10);
+    const inventory = React.createRef();
+    const dispatch = useDispatch();
+
+    const affectedSystems = useSelector(({ CVEStore }) =>CVEStore.affectedSystemsByCVE);
+
+    const handleSelect = (isChecked, payload) => {
+        if (!payload) { payload = []; }
+
+        setSelectedHosts(payload);
     };
 
-    constructor(props) {
-        super(props);
-        this.inventory = React.createRef();
-        this.state = {
-            InventoryCmp: () => <div>Loading...</div>,
-            removeListener: () => { },
-            StatusModal: () => null,
-            filter: undefined,
-            columns: [
-                {
-                    key: 'display_name',
-                    title: this.props.intl.formatMessage(messages.systemsColumnHeaderName),
-                    composed: ['facts.os_release', 'display_name'],
-                    props: {
-                        width: 40
-                    },
-                    renderFunc: (item, _id, { rule }) => renderSystemCSAwIcon(item, rule)
-                },
-                {
-                    key: 'status',
-                    title: this.props.intl.formatMessage(messages.status),
-                    props: {
-                        width: 30
-                    },
-                    renderFunc: (
-                        value,
-                        _id,
-                        { cve_status_id, status_id }
-                    ) => <StatusColumn cveStatusId={cve_status_id} systemStatusId={status_id} />
-                }
-            ],
-            selected: new Set(),
-            selectListener: middlewareListener.addNew({
-                on: 'SELECT_ENTITY',
-                callback: payload => this.onSelect(payload)
-            }),
-            selectAllListener: middlewareListener.addNew({
-                on: 'SELECT_ENTITIES',
-                callback: payload => this.onSelect(payload)
-            })
-        };
-    }
+    const onSelect = (payload) => {
+        const newSelected = payload.data.selected ? [payload.data.id, ...(selectedHosts || [])]
+            : selectedHosts && selectedHosts.filter(item => item !== payload.data.id);
+        payload.preventBubble();
```

Comment:
out of curiosity, why `preventBubble` ?

Reply:
middleware listener fires SELECT_ENTITY event 5 times when user selects a single row. I thought it would prevent it. But when I checked now, I realized that it does not.